util/reflect_test.go

```diff
@@ -847,3 +958,399 @@ func TestStructValueHasNFields(t *testing.T) {
+								Dir: map[string]*yang.Entry{
+									"inner": {
+										Name: "inner",
+										Kind: yang.DirectoryEntry,
+										Dir: map[string]*yang.Entry{
+											"leaf-field": {
+												Name: "leaf-field",
+												Kind: yang.LeafEntry,
+												Type: &yang.YangType{Kind: yang.Yint32},
+											},
+										},
+									},
+								},
+							},
+						},
+					},
+				},
+			},
+		},
+	}
+
+	c1 := &ContainerStruct1{
+		StructKeyList: map[string]*ListElemStruct1{
+			"forty-two": {
+				Key1:  String("forty-two"),
+				Outer: &OuterContainerType1{Inner: &InnerContainerType1{LeafName: Int32(1234)}},
+			},
+		},
+	}
+
+	tests := []struct {
+		desc       string
+		rootStruct interface{}
+		path       *gpb.Path
+		want       interface{}
+		wantErr    string
+	}{
+		{
+			desc:       "success leaf-ref",
+			rootStruct: c1,
+			path: &gpb.Path{
+				Elem: []*gpb.PathElem{
+					{
+						Name: "config",
+					},
+					{
+						Name: "simple-key-list",
+						Key: map[string]string{
+							"key1": "forty-two",
+						},
+					},
+					{
+						Name: "outer",
+					},
+					{
+						Name: "inner",
+					},
+					{
+						Name: "leaf-field",
+					},
+				},
+			},
+			want: []interface{}{c1.StructKeyList["forty-two"].Outer.Inner.LeafName},
+		},
+		{
+			desc:       "success leaf full path",
+			rootStruct: c1,
+			path: &gpb.Path{
+				Elem: []*gpb.PathElem{
+					{
+						Name: "config",
+					},
+					{
+						Name: "simple-key-list",
+						Key: map[string]string{
+							"key1": "forty-two",
+						},
+					},
+					{
+						Name: "outer",
+					},
+					{
+						Name: "config",
+					},
+					{
+						Name: "inner",
+					},
+					{
+						Name: "leaf-field",
+					},
+				},
+			},
+			want: []interface{}{c1.StructKeyList["forty-two"].Outer.Inner.LeafName},
+		},
+		{
+			desc:       "bad path",
+			rootStruct: c1,
+			path: &gpb.Path{
+				Elem: []*gpb.PathElem{
+					{
+						Name: "config",
+					},
+					{
+						Name: "simple-key-list",
+						Key: map[string]string{
+							"key1": "forty-two",
+						},
+					},
+					{
+						Name: "bad-element",
+					},
+					{
+						Name: "inner",
+					},
+					{
+						Name: "leaf-field",
+					},
+				},
+			},
+			want:    nil,
+			wantErr: `could not find path in tree beyond schema node simple-key-list, (type *util.ListElemStruct1), remaining path elem:<name:"bad-element" > elem:<name:"inner" > elem:<name:"leaf-field" > `,
+		},
+		{
+			desc:       "nil source field",
+			rootStruct: c1,
+			path: &gpb.Path{
+				Elem: []*gpb.PathElem{
+					{
+						Name: "config",
+					},
+					{
+						Name: "simple-key-list",
+						Key: map[string]string{
+							"key1": "forty-two",
+						},
+					},
+					{
+						Name: "outer2",
+					},
+					{
+						Name: "inner",
+					},
+					{
+						Name: "leaf-field",
+					},
+				},
+			},
+			want: []interface{}(nil),
+		},
+	}
+
+	for _, tt := range tests {
+		t.Run(tt.desc, func(t *testing.T) {
```

Comment:
For my education -- it seems like here you could run the anonymous function here directly within the test - with the only difference that it wouldn't run in its own goroutine. Is there another motivating factor for using `t.Run` here?